src/scripts/main.js

```diff
@@ -2,4 +2,22 @@
 
 document.addEventListener('click', (e) => {
   // write code here
+  e.preventDefault();
+
+  const wall = document.querySelector('.wall');
+  const spider = document.querySelector('.spider');
+  const spiderCentered = spider.clientWidth / 2;
+
+  let x = e.offsetX - spiderCentered;
+  let y = e.offsetY - spiderCentered;
+
+  x = Math.max(0, Math.min(x, wall.clientWidth - spider.clientWidth));
+  y = Math.max(0, Math.min(y, wall.clientHeight - spider.clientHeight));
+
+  if (!wall.contains(e.target)) {
```

Comment:
The condition `if (!wall.contains(e.target))` should be placed before calculating the `x` and `y` positions to avoid unnecessary calculations when the click is outside the wall.